### PR TITLE
Migrated from Docopt to ArgParse

### PR DESCRIPTION
**Pull request by calexandru2018**
- The CLI has been fully migrated to ArgParse
- CLI code has been written in OO paradigm
- None of the commands have been changed, fully backwards compatible with the previous version using Docopt

### CODE REVIEW

File: protonvpn_cli/cli.py [hunk 1]
```diff
@@ -88,13 +43,54 @@ def cli():
...
+            print(USAGE)
+            parser.exit()
+
+        getattr(self, args.command)()
+
+    # Intialize ProtonVPN profile
+    def init(self):
+        """Intialiazes ProtonVPN profile. To intialize profile inline, provide the "-i" option."""
+        parser = argparse.ArgumentParser(description="Initialize ProtonVPN profile", prog="protonvpn init")
+        parser.add_argument("-i", "--inline", nargs=3, required=False, help="Inline intialize profile. (username password protocol)", metavar="")
```
Comment (calexandru2018) [inline protonvpn_cli/cli.py:77]: This line is being flagged as too long since it is >120 chars long

File: protonvpn_cli/cli.py [hunk 2]
```diff
@@ -106,45 +102,95 @@ def cli():
...
+        parser = argparse.ArgumentParser(description="Connect to ProtonVPN", prog="protonvpn c")
+        group = parser.add_mutually_exclusive_group()
+        group.add_argument("servername", nargs="?", help="Servername (CH#4, CH-US-1, HK5-Tor).", metavar="")
+        group.add_argument("-f", "--fastest", help="Connect to the fastest ProtonVPN server.", action="store_true")
+        group.add_argument("-r", "--random", help="Connect to a random ProtonVPN server.", action="store_true")
+        group.add_argument("--cc", help="Connect to the specified country code (SE, PT, BR, AR).", metavar="")
+        group.add_argument("--sc", help="Connect to the fastest Secure-Core server.", action="store_true")
+        group.add_argument("--p2p", help="Connect to the fastest torrent server.", action="store_true")
+        group.add_argument("--tor", help="Connect to the fastest Tor server.", action="store_true")
+        parser.add_argument("-p", "--protocol", help="Connect via specified protocol.", choices=["udp", "tcp"], metavar="")
```
Comment (calexandru2018) [inline protonvpn_cli/cli.py:114]: This line is being flagged as too long since it is >120 chars long

File: protonvpn_cli/cli.py [hunk 3]
```diff
@@ -283,12 +329,12 @@ def print_examples():
...
         "               Connect to the fastest VPN Server.\n\n"
-        "protonvpn connect --cc AU\n"
+        "protonvpn connect -cc AU\n"
         "               Connect to the fastest Australian server\n"
         "               with the default protocol.\n\n"
-        "protonvpn c --p2p -p tcp\n"
+        "protonvpn c -p2p -p tcp\n"
         "               Connect to the fastest torrent server with TCP.\n\n"
-        "protonvpn c --sc\n"
+        "protonvpn c -sc\n"
```
Comment (Rafficer) [inline protonvpn_cli/cli.py:337]: Need to be reverted to `--`

File: protonvpn_cli/cli.py [hunk 4]
```diff
@@ -51,8 +6,8 @@
 import getpass
 import shutil
 import time
+import argparse
 # External Libraries
```
Comment (Rafficer) [inline protonvpn_cli/cli.py:10]: Leaving this comment is unnecessary, IMO.

**Comment by Rafficer** [comment]
docopt is still mentioned as dependency in setup.py and requirements.txt. 😜

**Comment by kaplun** [comment]
Hi! So since @calexandru2018 has implemented @Rafficer suggested changes and @ammirate has cross-checked, in order to unblock @calexandru2018 I dismiss the pending requested changes (since it's just a GitHub artifact at this point).

**Comment by iiriix** [comment]
The second to last commit was force-pushed and ignored this PR which was previously merged. I was wondering if that was intentional to revert this back? Getting back to Docopt again?

**Comment by kaplun** [comment]
We are preparing the official linux CLI (currently on beta: https://protonvpn.com/blog/linux-vpn-cli-beta/). For technical reasons (as we need to prepare for smooth update from this CLI to the official one) we needed to make a small technical release as part of the planned upgrade. Unfortunately master branch already had this docopt-to-argparse branch merged in, but was too risky to make the technical release to also include this branch.

**Pull request merged**